src/CachePlugin.php

```diff
@@ -69,15 +69,42 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
         $cacheItem = $this->pool->getItem($key);
 
         if ($cacheItem->isHit()) {
-            // return cached response
             $data = $cacheItem->get();
-            $response = $data['response'];
-            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+            if (isset($data['expiresAt']) && time() < $data['expiresAt']) {
```

Comment:
Hm. Removing the isset in 2.0 will not help when we fetch a stored v1 response and try to fetch it as v2. 

We could add a cache busting parameter when we create the cache key. 
Or we could make sure to always run isset even on 'response' and 'body'